packages/core/src/shared/utilities/searchUtils.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Performs a binary search on a sorted array of numbers to find the index of a target value.
+ *
+ * @param items The sorted array of numbers to search.
+ * @param target The target value to search for.
+ * @returns The index of the target value in the array, or -1 if not found.
+ */
+export function binarySearch(items: number[], target: number): number {
```

Comment:
why do we need binary search ?

Reply:
In the main Crash Reporting PR we get a list of PIDs which can be in the thousands and then search it to see if the ExtHost process is missing (implies a crash). And we do this for each running extension instance. So I thought we could be a bit more efficient about it.